lib/mutant/cli.rb

```diff
@@ -86,7 +86,7 @@ def parse(arguments)
     def parse_match_expressions(expressions)
       fail Error, 'No expressions given' if expressions.empty?
 
-      expressions.map(&Expression.method(:parse)).each do |expression|
+      expressions.map(&config.expression_parser.method(:call)).each do |expression|
```

Comment:
Yeah, I like that. I wish we'd have first class function composition to make this thing a one liner.